src/captcha/captcha.ts

```diff
@@ -71,6 +71,45 @@ export function parseCaptchaSolutions(captchaJSON: JSON): CaptchaSolution[] {
     }
 }
 
+function captchaSort<T extends { captchaId: string }>(a: T, b: T) {
+    return a.captchaId.localeCompare(b.captchaId);
+}
+
+async function sortAndComputeHashes(
+    received: CaptchaSolution[],
+    stored: Captcha[]
+): Promise<{ captchaId: string; hash: string }[]> {
+    const _received = received.sort(captchaSort);
+    const _stored = stored.sort(captchaSort);
+
+    return Promise.all(
+        _stored
+            .filter(({ solved }) => solved)
```

Comment:
This filter breaks the algorithm as it messes up the indexes between `_received` and `_stored`.

You have to compare all received versus all stored and calculate the hash for each received captcha, even if the captcha is already solved.

Reply:
Unit tests would help to catch this kind of thing.

Reply:
I've fixed this.